Gnome 45

nater1983 opened the pull request:
Here is a start of the Gnome 45 ESM Port

nater1983 committed:
Update extension.js

nater1983 committed:
Update helpers.js

nater1983 committed:
Update metadata.json

nater1983 commented:
I still get an error  ```assignment to undeclared variable day```

nater1983 committed:
Update helpers.js

Removed GLib.free(namedaysFile);

nater1983 committed:
Update helpers.js

Declare day and month variables

nater1983 committed:
Update helpers.js

Completed the Port to Gnome 45
Please review..

nater1983 commented:
Welp shutdown my VM and went back to it and came up with an error about an Array.

nater1983 committed:
Update extension.js

When disabled currentDateTime = null

nater1983 committed:
Update helpers.js

Cleaning Gio.File.new_for_uri(import.meta.url) when you need the current file,

nater1983 committed:
Update helpers.js

Fixed Expected type string for argument 'filename'  but got type GOject_Object

nater1983 committed:
Update extension.js

Correct currentDatetime to currentDateTime

nater1983 committed:
Update helpers.js

Correct currentDatetime to currentDateTime

nater1983 committed:
Update helpers.js

Sorry I should have just left this.

nater1983 committed:
Update helpers.js

nater1983 committed:
Update extension.js

nater1983 committed:
Update helpers.js

This change doesn't seem to affect the extension from working correctly.

nater1983 commented:
i have been running this updated version of eortologio with out any problems the last 2-3 days.

nater1983 commented:
@danchris just wondering if you have been able to take a look at this MR??

danchris commented:
Hi @nater1983, I haven't had time to see it, my schedule is quite full. I want some time to take a deep look to the ESM changes. Also, I want to setup a new environment for developing the extension. 

Thank you!

nater1983 commented:
understood...

danchris commented on eortologio@danchris.github.io/extension.js:
```diff
@@ -16,31 +16,27 @@
-
-let eortologioPopup;
-let currentDatetime;
+ let eortologioPopup;
```
Please, remove unused variable.

nater1983 replied:
Resolved with this commit https://github.com/danchris/eortologio-gnome-shell-extension/pull/5/commits/a705e30aeb458bb60aa7805e6c85a3b6bad5e30e

danchris commented on eortologio@danchris.github.io/helpers.js:
```diff
@@ -1,69 +1,70 @@
 
-function getRecurringNameDays(date){
+export function getRecurringNameDays(date, subdir){
+    const file = Gio.File.new_for_uri(import.meta.url).get_parent().get_child('recurring_namedays.json'); 
```
Here you create a pointer. Please, consider to free the memory properly before return from the function.

nater1983 replied:
per andyholmes he said we are not supposed to free the memory properly before the return from the function.

nater1983 replied:
We are just assigning a value to a variable, when the scope ends it will be collected (i.e. freed)

danchris commented on eortologio@danchris.github.io/helpers.js:
```diff
@@ -1,69 +1,70 @@
-   
-    let filePath = Me.dir.get_child('relative_to_easter.json').get_path();
-  
+    const file = Gio.File.new_for_uri(import.meta.url).get_parent().get_child('relative_to_easter.json'); 
```
Here you create a pointer. Please, consider to free the memory properly before return from the function.

nater1983 replied:
Understood I will work on this

nater1983 replied:
per andyholmes he said we are not supposed to free the memory properly before the return from the function.

nater1983 replied:
We are just assigning a value to a variable, when the scope ends it will be collected (i.e. freed)

nater1983 committed:
Update extension.js

removed let EortologioPopup; and updated all eortologioPopup to EortologioPopup

nater1983 commented:
Please review comments https://github.com/danchris/eortologio-gnome-shell-extension/pull/5#discussion_r1359660383 & https://github.com/danchris/eortologio-gnome-shell-extension/pull/5#discussion_r1359660398

danchris reviewed (approved):
LGTM

danchris commented:
Hi @nater1983 , thank you for your contribution!!
